src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4231,7 +4231,7 @@ public bool ConnectAsync(SocketAsyncEventArgs e)
                     {
                         InternalBind(new IPEndPoint(IPAddress.Any, 0));
                     }
-                    else
+                    else if (endPointSnapshot.AddressFamily == AddressFamily.InterNetworkV6)
```

Comment:
> Please explain how this is breaking app-compat with Desktop.

Instead of performing Bind using IPv6Any for other types of sockets (which may have not be correct in the first place), now it's only doing it for IPv4/v6. Regardless of code correctness, your change has no effect if the user wants to use the Desktop-based implementation in their `project.json` file.

> The only one now is Unix Domain Socket

From MSDN, AddressFamily already exposes support for a lot more other types of Socket types: https://msdn.microsoft.com/en-us/library/system.net.sockets.addressfamily(v=vs.110).aspx